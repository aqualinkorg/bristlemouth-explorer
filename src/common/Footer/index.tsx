import {
  AppBar,
  Link,
  Stack,
  Typography,
  styled,
  useMediaQuery,
} from '@mui/material';
import aqualinkLogo from 'src/assets/aqualink-logo.png';
import {
  aqualinkURL,
  bristlemouthURL,
  bristlemouthExplorerGithub,
} from 'src/helpers/constants';

const StyledAppBar = styled(AppBar)(({ theme }) => ({
  top: 'auto',
  bottom: 0,
  backgroundColor: theme.palette.dark?.main,
  color: 'white',
}));

const StackContainer = styled(Stack)(({ theme }) => ({
  justifyContent: 'center',
  alignItems: 'center',
  padding: theme.spacing(2),
  [theme.breakpoints.down('md')]: {
    flexDirection: 'column',
    gap: '1rem',
    position: 'relative',
  },
}));

const StackAqualinkLogo = styled(Stack)(({ theme }) => ({
  display: 'flex',
  flexWrap: 'nowrap',
  justifyContent: 'center',
  alignItems: 'center',
  gap: theme.spacing(1),
}));

const Logo = styled('img')(({ theme }) => ({
  borderRadius: theme.spacing(1),
  height: '2.5rem',
}));

const StyledLink = styled(Link)(() => ({
  textDecoration: 'none',
}));

const ContactUsTypography = styled(Typography)(({ theme }) => ({
  textAlign: 'right',
  position: 'absolute',
  top: theme.spacing(2),
  right: theme.spacing(2),
  [theme.breakpoints.down('md')]: {
    position: 'initial',
    textAlign: 'center',
  },
}));

const BristlemouthLinkTypography = styled(Typography)(({ theme }) => ({
  position: 'absolute',
  left: theme.spacing(2),
  display: 'flex',
  [theme.breakpoints.down('md')]: {
    position: 'initial',
  },
}));

function Footer() {
  const isMobile = useMediaQuery('(max-width: 760px)');
  return (
    <StyledAppBar position={isMobile ? 'relative' : 'fixed'}>
      <StackContainer direction="row">
        <BristlemouthLinkTypography>
          <StyledLink
            target="_blank"
            rel="noopener noreferrer"
            href={bristlemouthURL}
          >
            Bristlemouth.org
          </StyledLink>
          <StyledLink
            target="_blank"
            rel="noopener noreferrer"
            href={bristlemouthURL}
            style={{
              marginLeft: '2rem',
              fontWeight: 'bold',
              textAlign: 'center',
            }}
          >
            Apply for the pioneer program
          </StyledLink>
        </BristlemouthLinkTypography>

        <StackAqualinkLogo direction="row">
          <Typography textAlign="center">Developed with ❤️ by</Typography>
          <Link target="_blank" rel="noopener noreferrer" href={aqualinkURL}>
            <Logo src={aqualinkLogo} alt="Aqualink logo" />
          </Link>
        </StackAqualinkLogo>

        <ContactUsTypography>
          Bristlemouth Explorer is open source!
          <br />
          Find us on{' '}
          <StyledLink
            target="_blank"
            rel="noopener noreferrer"
            href={bristlemouthExplorerGithub}
          >
            GitHub
          </StyledLink>
        </ContactUsTypography>
      </StackContainer>
    </StyledAppBar>
  );
}

export default Footer;
