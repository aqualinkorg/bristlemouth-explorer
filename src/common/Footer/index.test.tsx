import { render } from '@testing-library/react';
import Footer from '.';

it('renders as expected', () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});
