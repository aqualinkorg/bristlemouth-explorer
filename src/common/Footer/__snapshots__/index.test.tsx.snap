// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1o4jfbn-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiStack-root css-kzjm6r-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-14ylj2n-MuiTypography-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-o4qrxl-MuiTypography-root-MuiLink-root"
          href="https://www.bristlemouth.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Bristlemouth.org
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-o4qrxl-MuiTypography-root-MuiLink-root"
          href="https://www.bristlemouth.org/"
          rel="noopener noreferrer"
          style="margin-left: 2rem; font-weight: bold; text-align: center;"
          target="_blank"
        >
          Apply for the pioneer program
        </a>
      </p>
      <div
        class="MuiStack-root css-17fi1qs-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
        >
          Developed with ❤️ by
        </p>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          href="https://aqualink.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Aqualink logo"
            class="css-4vnfkb"
            src="[object Object]"
          />
        </a>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-jgunqj-MuiTypography-root"
      >
        Bristlemouth Explorer is open source!
        <br />
        Find us on
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-o4qrxl-MuiTypography-root-MuiLink-root"
          href="https://github.com/aqualinkorg/bristlemouth-explorer"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </p>
    </div>
  </header>
</div>
`;
